Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppDeploy/IISWebAppDeployV1/DeployIISWebApp.ps1

```diff
@@ -40,6 +41,19 @@ function Get-ScriptToRun
     $websiteName = EscapeSpecialChars -str $websiteName
     $AdditionalArguments = EscapeSpecialChars -str $AdditionalArguments
 
+    $useSanitizerCall = Get-SanitizerCallStatus
+    $useSanitizerActivate = Get-SanitizerActivateStatus
+
+    if ($useSanitizerCall) 
+    {
+        $sanitizedArguments = Protect-ScriptArguments -InputArgs $AdditionalArguments -TaskName "IISWebAppDeployV1"
+    }
+
+    if ($useSanitizerActivate) 
+    {
+        $AdditionalArguments = $sanitizedArguments
```

Comment:
Looks like sanitizedArguments won't have a value if useSanitizerCall == false. maybe it's worth to merge useSanitizerCall and useSanitizerActivate as one flag, no?

Reply:
Got it, we need telemetry side effect of Protect-ScriptArguments even when useSanitizerActivate == false